extendeddatasquare.go

```diff
@@ -15,29 +15,39 @@ import (
 type ExtendedDataSquare struct {
 	*dataSquare
 	codec             Codec
+	treeName          string
 	originalDataWidth uint
 }
 
 func (eds *ExtendedDataSquare) MarshalJSON() ([]byte, error) {
 	return json.Marshal(&struct {
 		DataSquare [][]byte `json:"data_square"`
 		Codec      string   `json:"codec"`
+		Tree       string   `json:"tree"`
 	}{
 		DataSquare: eds.dataSquare.Flattened(),
 		Codec:      eds.codec.Name(),
+		Tree:       eds.treeName,
 	})
 }
 
 func (eds *ExtendedDataSquare) UnmarshalJSON(b []byte) error {
 	var aux struct {
 		DataSquare [][]byte `json:"data_square"`
 		Codec      string   `json:"codec"`
+		Tree       string   `json:"tree"`
 	}
 
 	if err := json.Unmarshal(b, &aux); err != nil {
 		return err
 	}
-	importedEds, err := ImportExtendedDataSquare(aux.DataSquare, codecs[aux.Codec], NewDefaultTree)
+
+	treeConstructor, err := TreeFn(aux.Tree)
+	if err != nil {
+		return err
```

Comment:
[question] I think this is still breaking due to the new `Tree` field in the JSON serialization of EDS. If an EDS doesn't have a Tree field (i.e. "") then I think the unmarshal will fail here. 

Is that correct? I'm mostly just clarifying that we should retain the breaking `!` in the PR title and commit history.

Reply:
> JSON breakage is not that concerning because it's not widely used. We don't store it anywhere and only use it for RPC. On golang side it will atomically fetch changes with an update, and we will need to notify the rust client about such changes

Related comment I had.

Reply:
I think keeping ! is correct because it's breaking

Reply:
Thanks! Resolving.